WIP: Pass cwd for hooks exec run function

fangpenlin opened the pull request:
#### What type of PR is this?

<!--
Please label this pull request according to what type of issue you are
addressing, especially if this is a release targeted pull request.

Uncomment only one `/kind <>` line, hit enter to put that in a new line, and
remove leading whitespace from that line:
-->

For fixing the wrong cwd bug in podman:

https://github.com/containers/podman/issues/18907

> /kind bug

#### What this PR does / why we need it:

I added a new `cwd` argument for run hooks in common package:

https://github.com/containers/common/pull/1514

for fixing cwd not set bug of poststop hook exec run in podman:

https://github.com/containers/podman/issues/18907

It seems like buildah also depends on the same function and also call hooks exec, so this PR fixes the issue altogether.

#### How to verify it
Run unit tests

#### Which issue(s) this PR fixes:

Fixes https://github.com/containers/podman/issues/18907

<!--
Automatically closes linked issue when PR is merged.
Uncomment the following comment block and include the issue
number or None on one line.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`, or `None`.
-->

<!--
Fixes #
or
None
-->

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes please follow the kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note

```

TomSweeneyRedHat commented:
Now that https://github.com/containers/common/pull/1521 has merged, are there adjustments necessary for this PR?

fangpenlin commented:
@TomSweeneyRedHat yeah, changes are needed, I will update it later tonight

fangpenlin commented:
@TomSweeneyRedHat I have updated the PR with the new method I added in common lib.
But it seems like a new release was not made yet from there

https://github.com/containers/common/pull/1524

I guess I can only wait until they merge it and make a new one then? By the way, should I check-in vendor changes in my PR? I saw that they were not excluded from the Git repo

rhatdan commented:
@fangpenlin This needs a rebase.

fangpenlin commented:
@rhatdan Rebased. But I saw there's a `//nolint:staticcheck` comment seems like disabling lint check or what? Not sure what's it for, should I add the same thing to my changes?

rhatdan commented:
I would only worry about the lint check if lint is blowing up. Looks like tests are failing though.

TomSweeneyRedHat commented:
@fangpenlin c/common v0.54.0 has been created fwiw.  I'm trying to vendor it now into Buildah, but I'm having some machine issues. doing so.

rhatdan commented:
@fangpenlin Still working on this?

fangpenlin commented:
Oh, sorry, forgot this one. Just rebased

rhatdan commented:
Remove [NO NEW TESTS NEEDED]  from commit message, you should be all set then.

fangpenlin commented:
Okay, done. What about now?

fangpenlin commented:
@rhatdan the Smoke Test says this

```
pr-should-include-tests: PR does not include changes in the 'tests' directory
```

Should I add the `[NO NEW TESTS NEEDED]` back and make the comment shorter?

fangpenlin commented:
@flouthoc 

I am not too familiar with buildah code base. But I did a quick search and it seems like the `Run` method is called from the paths of major buildah functions, like `from`, `run` and others:

https://github.com/containers/buildah/blob/8aa4d360e7b388477153e94b32725ff50055ce39/cmd/buildah/from.go#L169
https://github.com/containers/buildah/blob/8aa4d360e7b388477153e94b32725ff50055ce39/cmd/buildah/run.go#L182
https://github.com/containers/buildah/blob/8aa4d360e7b388477153e94b32725ff50055ce39/imagebuildah/stage_executor.go#L661

So I guess as long as the existing tests run very basic building image stuff, that path should be touched at some point. The only thing missing is that we didn't test specifically for the cwd part. Personally I would rely more on the test cases from upstream project `common` for this case

https://github.com/containers/common/blob/74c49fcec3ef8a08d0437eea3fea6eb20787a33a/pkg/hooks/exec/exec_test.go#L146-L166

However, if a specific tests are really needed, I can find a time to add it. Maybe test against `setupOCIHooks`? The code lives in `run_linux.go`, looks like linux specific implementation. Not sure how to write test for it tho, might need to look it up a bit more for that part.
